### DISCUSSION TITLE
EPSG:4326 projection method, how to set the viewport

### DISCUSSION THREAD

**Discussion by shaojun427** [visgl/deck.gl#5513] [Q&A]
![image](https://user-images.githubusercontent.com/16127112/108715440-d7e96800-7555-11eb-8c2b-624d7677d2a2.png)
I have modified the projection method of mapbox-gl, the slice rendering is normal, but the viewport of the deck is Mercator projection by default. If I want to support other EPSG, do I need to configure it based on the basic viewport?

**Comment by Pessimistress** [comment]
You can use the [OrthographicView](https://deck.gl/docs/api-reference/core/orthographic-view).

**Reply by Emelia-L** [reply]
I have the same question too.  I have got 4326 tiles now, but the projection of Deck with mapview is still 3857. So the position can not be consistent with 4326 map tiles. If I want to change Deck projection to show correct data on 4326 map tiles, what should I do?

**Reply by kylebarron** [reply]
The `TileLayer` currently only supports Web Mercator tiles in the OSM indexing system. There's some discussion here about making it easier for users to extend the `TileLayer` to more tiling systems https://github.com/visgl/deck.gl/pull/5504

**Reply by Pessimistress** [reply]
> The TileLayer currently only supports Web Mercator tiles in the OSM indexing system

The TileLayer supports non-geospatial views also, as shown in https://deck.gl/examples/tile-layer-non-geospatial/

You just need to manually set the [extent](https://deck.gl/docs/api-reference/geo-layers/tile-layer#extent) prop to the geo bounding box (`[-180, -90, 180, 90]`).

**Reply by ibgreen** [reply]
@math.gl/proj4 module supports reprojection from WKT-CRS coordinate systems to e.g web mercator. Perhaps it is possible to add a coordinate post processing step in the onTileLoad function?

**Reply by Pessimistress** [reply]
@ibgreen I believe the original ask was to plot the coordinates as is and not use Web Mercator projection, which works out of the box with OrthographicView: https://gist.github.com/tech-meppem/5a40e00b04220e924177791b4982be0f

**Reply by kylebarron** [reply]
My (possibly incorrect) interpretation, especially of @Emelia-L 's comment, is that the user has geospatial data that they want to use with a geospatial view and possibly a basemap, but the data are tiled in the lon/lat tiling system, which doesn't work with the existing `TileLayer`.

> Perhaps it is possible to add a coordinate post processing step in the onTileLoad function?

If source data is _tiled_ according to a tiling system other than the OSM indexing system, then this is not possible, because for any `x-y-z` index, the OSM system puts it at a different location on the globe than another tiling system. That's the point of the discussion in the linked PR https://github.com/visgl/deck.gl/pull/5504
